eodag/api/search_result.py

```diff
@@ -42,6 +47,43 @@ def crunch(self, cruncher, **search_params):
         crunched_results = cruncher.proceed(self, **search_params)
         return SearchResult(crunched_results)
 
+    def filter_date(self, start=None, end=None):
+        """Apply FilterDate crunch, check its documentation to know more"""
+        return self.crunch(FilterDate(dict(start=start, end=end)))
+
+    def filter_latest_intersect(self, geometry):
+        """Apply FilterLatestIntersect crunch, check its documentation to know more"""
+        return self.crunch(FilterLatestIntersect({}), geometry=geometry)
+
+    def filter_latest_by_name(self, name_pattern):
+        """Apply FilterLatestByName crunch, check its documentation to know more"""
+        return self.crunch(FilterLatestByName(dict(name_pattern=name_pattern)))
+
+    def filter_overlap(
+        self,
+        geometry,
+        minimum_overlap=None,
+        contains=None,
+        intersects=None,
+        within=None,
```

Comment:
```suggestion
        minimum_overlap=0,
        contains=False,
        intersects=False,
        within=False,
```

Reply:
My reasoning here is the same as for `filter_property`